File: MyApplication2/app/src/main/java/com/example/mengzhi/myapplication2/MainActivity.java
package com.example.mengzhi.myapplication2;

import android.animation.Animator;
import android.animation.AnimatorInflater;
import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.app.Activity;
import android.graphics.Color;
import android.graphics.drawable.AnimationDrawable;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.animation.AnimationUtils;
import android.view.animation.ScaleAnimation;
import com.example.mengzhi.myapplication2.EnvUtil;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.hitouch_version);
        View view = findViewById(R.id.version);

        ObjectAnimator animEnter = (ObjectAnimator) AnimatorInflater.loadAnimator(this, R.animator.anim_enter);
        ObjectAnimator animExit = (ObjectAnimator) AnimatorInflater.loadAnimator(this, R.animator.anim_exit);
        ObjectAnimator animShow = (ObjectAnimator) AnimatorInflater.loadAnimator(this, R.animator.anim_show);
        ObjectAnimator animHide = (ObjectAnimator) AnimatorInflater.loadAnimator(this, R.animator.anim_hide);

        ScaleAnimation animScaleSmall = (ScaleAnimation) AnimationUtils.loadAnimation(this, R.anim.anim_scale_small);
        ScaleAnimation animScaleLarge = (ScaleAnimation) AnimationUtils.loadAnimation(this, R.anim.anim_scale_large);
        view.startAnimation(animScaleSmall);
        view.startAnimation(animScaleLarge);

        view.setBackgroundResource(R.drawable.trans_color);
        AnimationDrawable animationDrawable = (AnimationDrawable) view.getBackground();
        animationDrawable.setOneShot(false);
        animationDrawable.start();

        AnimatorSet animatorSet = new AnimatorSet();
        animatorSet.play(animHide).after(500L).after(animShow);
        animatorSet.setTarget(view);
        animatorSet.start();

//        AnimatorSet animatorSet2 = new AnimatorSet();
//        animatorSet2.play(animExit).after(5000L).after(animEnter);
//        animatorSet2.setTarget(view);
//        animatorSet2.start();

        Log.d("zhimeng", "BuildConfig.DEBUG: " + BuildConfig.debug);
    }
}

File: MyApplication2/app/src/main/java/com/example/mengzhi/myapplication2/MainActivity.kt
package com.example.mengzhi.myapplication2

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import com.example.mengzhi.annotationTesting.InjectUtils
import com.example.mengzhi.annotationTesting.OnClick
import com.example.mengzhi.iotesting.Course
import com.example.myarouter_annotations.MyARouter
import com.example.myarouterapi.ParameterManager
import com.example.myarouterapi.RouterManager

@MyARouter(path = "/app/MainActivity")
class MainActivity : Activity() {
    private lateinit var testParcelButton: Button
    private lateinit var showAnimActivityButton: Button

//    @MyParam
//    private lateinit var orderDrawable: OrderDrawable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main3)
    }

    override fun onResume() {
        testParcelButton = findViewById(R.id.parcel_btn)
        showAnimActivityButton = findViewById(R.id.anim_btn)
        showAnimActivityButton.setOnClickListener(getAnimBtnListener())
        testParcelButton.setOnClickListener {
//            getParcelBtnListener()
        }
        super.onResume()

        InjectUtils.injectEvent(this)

//        ParameterManager.getInstance()?.loadParameter(this)
    }

//    @OnClick([R.id.anim_btn])
    fun onClick(view: View): Boolean {
        getAnimBtnListener()
        return false
    }

    @OnClick([R.id.start_service_btn])
    fun startServiceOnClick(view: View) {
        Log.i(TAG, "click startServiceOnClick")
        val intent = Intent()
        intent.action = "com.eample.dream.SERVICE"
        intent.`package` = "com.example.mengzhi.myapplication2"
        startService(intent)
    }

    private fun getCourse(): Course {
        val course = Course()
        course.name = "Computer Organization"
        course.score = 100
        course.studentList = arrayListOf("xiao minng", "li hua")
        return course
    }

    private fun getAnimBtnListener(): View.OnClickListener {
        return View.OnClickListener {
//            val intent = Intent()
//            intent.setClass(this, AnimTestActivity::class.java)
//            intent.putExtra("course", getCourse())
//            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
//            startActivity(intent)
            RouterManager.getInstance()
                ?.build("/app/AnimTestActivity")
                ?.withString("name", "hello this is MainActivity")
                ?.navigation(this)
            finish()
        }
    }

    companion object {
        private const val TAG = "MainActivity"
    }
}